File: app/src/jvx/java/art/coded/wireframe/view/adapter/PagingAdapter.java
package art.coded.wireframe.view.adapter;

import android.app.Activity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.core.app.ShareCompat;
import androidx.paging.PagedListAdapter;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.RecyclerView;

import art.coded.wireframe.R;
import art.coded.wireframe.model.entity.Element;

public class PagingAdapter extends PagedListAdapter<Element, PagingAdapter.ElementViewHolder> {

    private static final String LOG_TAG = PagingAdapter.class.getSimpleName();

    LayoutInflater mLayoutInflater;
    Activity mActivity;

    public PagingAdapter(@NonNull DiffUtil.ItemCallback<Element> diffCallback, Activity activity) {
        super(diffCallback);
        mActivity = activity;
        mLayoutInflater = LayoutInflater.from(mActivity);
    }

    @NonNull @Override public ElementViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = mLayoutInflater.inflate(R.layout.list_item, parent, false);
        return new PagingAdapter.ElementViewHolder(itemView, mActivity);
    }

    @Override public void onBindViewHolder(@NonNull ElementViewHolder holder, int position) {
        Element element = getItem(position);
        holder.mItemView.setText(element == null ? "No elements" : element.getName());
    }

    static class ElementViewHolder extends RecyclerView.ViewHolder {

        private final TextView mItemView;
        private final ImageButton mImageButton;

        public ElementViewHolder(@NonNull View itemView, Activity activity) {

            super(itemView);
            mItemView = itemView.findViewById(R.id.textView);
            mImageButton = itemView.findViewById(R.id.list_button_share);
            mImageButton.setOnClickListener(view -> {

                ShareCompat.IntentBuilder
                        .from(activity)
                        .setType("text/plain")
                        .setText(mItemView.getText())
                        .startChooser();
            });
        }
    }
}

File: app/src/ktx/java/art/coded/wireframe/view/adapter/PagingAdapter.kt
package art.coded.wireframe.view.adapter

import androidx.recyclerview.widget.DiffUtil
import android.app.Activity
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import art.coded.wireframe.R
import android.widget.TextView
import android.widget.ImageButton
import androidx.core.app.ShareCompat
import androidx.paging.PagedListAdapter
import android.view.View
import art.coded.wireframe.model.entity.Element

private val LOG_TAG = PagingAdapter::class.java.simpleName

class PagingAdapter(diffCallback: DiffUtil.ItemCallback<Element>, val activity: Activity) :
    PagedListAdapter<Element, PagingAdapter.ElementViewHolder>(diffCallback) {
    private val layoutInflater: LayoutInflater = LayoutInflater.from(activity)
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ElementViewHolder {
        val itemView = layoutInflater.inflate(R.layout.list_item, parent, false)
        return ElementViewHolder(itemView, activity)
    }

    override fun onBindViewHolder(holder: ElementViewHolder, position: Int) {
        val element = getItem(position)
        holder.textView.text = element?.name ?: "No elements"
    }

    class ElementViewHolder(itemView: View, activity: Activity?) :
        RecyclerView.ViewHolder(itemView) {
        val textView: TextView = itemView.findViewById(R.id.textView)
        private val imageButton: ImageButton = itemView.findViewById(R.id.list_button_share)

        init {
            imageButton.setOnClickListener { view: View ->
                ShareCompat.IntentBuilder
                    .from(activity!!)
                    .setType("text/plain")
                    .setText(textView.text)
                    .startChooser()
            }
        }
    }
}